uc-crux-llvm/src/UCCrux/LLVM/Config/FromEnv.hs

```diff
@@ -58,13 +68,34 @@ suggestedLoopBound = 8
 suggestedRecursionBound :: Word64
 suggestedRecursionBound = 8
 
+-- | Parse (as in \"parse, don\'t validate\") options gathered by Crux
+-- ('UCCruxLLVMOptions') into an 'AppContext' and a 'TopLevelConfig'.
 processUCCruxLLVMOptions ::
-  (CruxOptions, UCCruxLLVMOptions) -> IO (AppContext, CruxOptions, UCCruxLLVMOptions)
+  (CruxOptions, UCCruxLLVMOptions) -> IO (AppContext, CruxOptions, TopLevelConfig)
 processUCCruxLLVMOptions (initCOpts, initUCOpts) =
   do
     let appCtx = makeAppContext (verbosityFromInt (verbosity initUCOpts))
-    when (not (doExplore initUCOpts) && null (entryPoints initUCOpts)) $
-      die "At least one entry point (--entry-points) is required (or try --explore)"
+    let doCrashEquivalence = crashEquivalence initUCOpts /= ""
+
+    entries <-
+      if doExplore initUCOpts
+      then pure Nothing
+      else
+        if doCrashEquivalence
+        then pure (nonEmpty (entryPoints initUCOpts))
+        else
+          Just <$>
+            maybe
+              (die
+                (unwords
+                  [ "At least one entry point (--entry-points) is required",
+                    "(or try --explore or --crash-equivalence)"
+                  ]))
+              pure
+              (nonEmpty (entryPoints initUCOpts))
```

Comment:
There is a substantial amount of logic in this line - can we wrap this up as a function with a comment describing what the idea is at a high level?

Reply:
Done!